http/handler_collection.go

```diff
@@ -43,6 +46,11 @@ func (s *collectionHandler) Create(rw http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	ctx := req.Context()
+	if req.Header.Get(DocEncryptionHeader) == "1" {
```

Comment:
todo: Since this header can be applied to any request this should be moved to a middleware.

Reply:
what do you mean by "any"? Are there headers that can't be applied?
This header is relevant only when creating docs. What's the point of moving it to a middleware?

Reply:
See my above comment about the query param.

Reply:
done